client/internal/receivedprocessor.go

```diff
@@ -52,9 +52,13 @@ func newReceivedProcessor(
 func (r *receivedProcessor) ProcessReceivedMessage(ctx context.Context, msg *protobufs.ServerToAgent) {
 	if r.callbacks != nil {
 		if msg.Command != nil {
-			r.rcvCommand(msg.Command)
-			// If a command message exists, other messages will be ignored
-			return
+			if r.hasCapability(protobufs.AgentCapabilities_AgentCapabilities_AcceptsRestartCommand) {
```

Comment:
Perhaps it would be useful to add a comment that says when any new command capabilities are added they need to be checked here. I am not sure what's the best way to ensure we don't forget to do it, but a comment may help.

Reply:
Added note